pkg/abstractions/image/build.go

```diff
@@ -282,14 +282,19 @@ func (b *Builder) Build(ctx context.Context, opts *BuildOpts, outputChan chan co
 	}
 
 	// Generate the commands to run in the container
+	pipCommands := []string{}
+	pipIndex := -1
 	for _, step := range opts.BuildSteps {
 		switch step.Type {
 		case shellCommandType:
 			opts.Commands = append(opts.Commands, step.Command)
 		case pipCommandType:
-			opts.Commands = append(opts.Commands, b.generatePipInstallCommand([]string{step.Command}, opts.PythonVersion))
+			opts.Commands = append(opts.Commands, "")
+			pipIndex = len(opts.Commands) - 1
+			pipCommands = append(pipCommands, step.Command)
 		}
 	}
+	opts.Commands[pipIndex] = b.generatePipInstallCommand(pipCommands)
```

Comment:
One of the benefits of `add_python_packages` was to be able to install packages before or after shell commands due to external dependency requirements. If we resolve and install all python packages in one go, this defeats the purpose of having an ordered list of commands.

Reply:
Just cause I'm curious - why do this in the first place? Was there a noticeable speed improvement? If yes, what was the duration differences between uv and uv+command grouping. If there's no particular reason, then this comment can be ignored.

Reply:
There is a noticeable speed improvement. But also, to clarify this still maintains the ordering. That is why I store the pip index and put an empty command in the command array. It is a bit fragile because if you have two different `add_python_packages` my assumptions fall apart. But as it is right now, instead of getting 

```
command 
command 
command
pip
pip
pip
``` 
you get 
```
command 
command 
command
pip
``` 

If the `add_python_packages` was put above the `add_commands` the order above would be reversed. The goal is not to do all commands and all pip together, but to do those listed together at once.

Reply:
The difference for uv was 4 seconds less when grouped. Not enormous, but not nothing

Reply:
So something like this

```
pip
command 
command 
command
pip
pip
```

would still do this?

```
pip
command 
command 
command
pip (grouped)
```

Reply:
Yeah exactly. Right now, my implementation is broken for the multiple pip case because I was just curious what performance would be like. But I think it might be worth doing it for real

Reply:
Okay perfect. Yeah I think we'd want this only if we can support using multiple calls to `add_python_packages`.

Reply:
Yeah I have to fix this in my conda PR actually. So I'll work out a way of supporting multiple calls.